blog/content/post/2020-04-29-trenchboot-uefi-environment.md

```diff
@@ -0,0 +1,833 @@
+          and how to install TrenchBoot on them.
+          
+cover: /covers/trenchboot-logo.png
+author: michal.zygowski
+layout: post
+published: true
+date: 2020-05-06
+archives: "2020"
+
+tags:
+  - firmware
+  - security
+  - open-source
+categories:
+  - Firmware
+  - OS Dev
+  - Security
+
+---
+
+If you haven't read previous blog posts about the TrenchBoot project, I
+encourage you to do so. We have presented our motivation, project's overview
+concept, and goals there. If you want to keep up with project development, stay
+tuned, and always try to catch up with missed articles.
+
+> Those articles will not always be strictly technical. Therefore, if you think
+some topics should be elaborated, feel free to ask us a question.
+
+In this article, I will explain how to install the NixOS on machines with UEFI
+firmware and how to install TrenchBoot. The most essential parts of the system
+will be also described in detail. It's important to understand their role, so
+you will be able to reproduce the environment smoothly and consciously. For now
+TrenchBoot focuses only on early launch (the initial launch of operating
+system), but the late launch (relaunching the operating system with DRTM
+without physical reboot) is also on the roadmap. Find out more about DRTM in the
+[TCG DRTM Architecture specification](https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf)
+
+## Testing platform
+
+As you already know, our solution is built for AMD processors. Therefore, for
+tests, we use suitable platforms. Our hardware infrastructure is still being
+expanded. Eventually, there will be a few more devices with different
+characteristics. For this particular article, we use **Supermicro Super**
+**Server M11SDV-4C** which has **AMD EPYC 3151** processor. We use connection
+via serial port, so every relevant output will be presented in the form of
+logs. What is important, our platform is equipped with **TPM** module (aka
+dTPM) which is obligatory in this project. Make sure you have it on your
+platform too.
+
+## Operating system
+
+To better understand the next section, first I will introduce an operating
+system which we use. Our choice is [NixOS](https://nixos.org/nixos/) which is a
+Linux distribution based on unusual (but reliable) package management. You can
+install tools from official packages or build your ones. The second feature
+will be strongly utilized in the entire project.
+
+> Of course, you can use any other Linux distribution. However, every procedure
+will be carried out by us in NixOS. If you won't have all dedicated tools and
+(supported and/or compatible) configuration, we can't provide full reliability.
+
+#### NixOS installation
+
+Now, I will guide through the installation of NixOS. After it, you should have
+an operating system in a default configuration. Later, I will show you how to
+customize it in runtime.
```

Comment:
@miczyg1 I am thinking about using `in` vs `at` vs `during`, tough choice

Reply:
`at runtime`?